### PR TITLE
JDK-8300133: Use generalized see and link tags in core libs

### PR DESCRIPTION
**Pull request by jddarcy**
With generalized see and link tags that can refer to anchors (JDK-8200337), the see and link tags in core libraries should be updated to use this feature when possible. This PR covers such updates for java.base.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8300133](https://bugs.openjdk.org/browse/JDK-8300133): Use generalized see and link tags in core libs

### Reviewers
 * [Jonathan Gibbons](https://openjdk.org/census#jjg) (@jonathan-gibbons - **Reviewer**) ⚠️ Review applies to [aa623f2a](https://git.openjdk.org/jdk/pull/12000/files/aa623f2a7a49f341acba748af67bc1db321a45b8)
 * [Mandy Chung](https://openjdk.org/census#mchung) (@mlchung - **Reviewer**)
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)
 * [Stuart Marks](https://openjdk.org/census#smarks) (@stuart-marks - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12000/head:pull/12000` \
`$ git checkout pull/12000`

Update a local copy of the PR: \
`$ git checkout pull/12000` \
`$ git pull https://git.openjdk.org/jdk pull/12000/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12000`

View PR using the GUI difftool: \
`$ git pr show -t 12000`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12000.diff">https://git.openjdk.org/jdk/pull/12000.diff</a>

</details>

### CODE REVIEW

**Review by jonathan-gibbons — APPROVED** [review]
Nice to see all those `<a href="{@docRoot}/..."` references go away.

File: src/java.base/share/classes/java/util/ResourceBundle.java [hunk 1]
```diff
@@ -267,8 +266,8 @@
  * A common format of resource bundles is in {@linkplain PropertyResourceBundle
  * .properties} file format.  Typically {@code .properties} resource bundles
  * are packaged in a JAR file.  Resource bundle only JAR file can be readily
- * deployed as an <a href="{@docRoot}/java.base/java/lang/module/ModuleFinder.html#automatic-modules">
- * automatic module</a>.  For example, if the JAR file contains the
+ * deployed as an {@linkplain java.lang.module.ModuleFinder##automatic-modules
+ * automatic module1}.  For example, if the JAR file contains the
```
Comment (naotoj) [inline src/java.base/share/classes/java/util/ResourceBundle.java:270]: Typo: extra `1`

Comment (mlchung) [inline src/java.base/share/classes/java/util/ResourceBundle.java:270]: typo : `module1`?

Comment (jddarcy) [reply src/java.base/share/classes/java/util/ResourceBundle.java]: Doh! Thanks; fixed in next push.

**Review by mlchung — APPROVED** [review]

**Review by naotoj — APPROVED** [review]
Changes in `ResoruceBundle` look good.

**Review by dfuch — COMMENTED** [review]
Changes to InetAddressResolverProvider.java look good.

File: src/java.base/share/classes/java/lang/CharSequence.java [hunk 2]
```diff
@@ -73,8 +73,8 @@ public interface CharSequence {
      * indexing.
      *
      * <p>If the {@code char} value specified by the index is a
-     * <a href="{@docRoot}/java.base/java/lang/Character.html#unicode">surrogate</a>, the surrogate
-     * value is returned.
+     * {@linkplain Character##unicode surrogate}, the surrogate value
```
[the remark this answers is not in the document]
Comment (RogerRiggs) [reply src/java.base/share/classes/java/lang/CharSequence.java:76]: The CSR has a concise description: [JDK-8294195](https://bugs.openjdk.org/browse/JDK-8294195) Generalize see and link tags for user-defined anchors

**Review by stuart-marks — APPROVED** [review]
Changes in collections look good. I think there are a bunch of other places where this could be used, but fixing up optional-restrictions is a good start.

**Pull request closed without merging**